### DISCUSSION TITLE
About asynchronous Logging Performance.

### DISCUSSION THREAD

**Discussion by tcmot** [apache/logging-log4j2#2107] [General]
Throughput performance test running the MultiThreadPerfTest and PerfTest.

`java -Dlog4j.configurationFile=log4j2.xml -DLog4jContextSelector=org.apache.logging.log4j.core.async.AsyncLoggerContextSelector org.apache.logging.log4j.core.async.perftest.PerfTest org.apache.logging.log4j.core.async.perftest.RunLog4j2`

`java -Dlog4j.configurationFile=log4j2.xml -DLog4jContextSelector=org.apache.logging.log4j.core.async.AsyncLoggerContextSelector org.apache.logging.log4j.core.async.perftest.MultiThreadPerfTest org.apache.logging.log4j.core.async.perftest.RunLog4j2`

```
log4j2.xml

<?xml version="1.0" encoding="UTF-8"?>
<Configuration>
  <Appenders>
    <RollingRandomAccessFile name="File" fileName="H:/log/app.log"
      filePattern="./log/app-%d{yyyy-MM-dd HH-mm-ss}-.log">
      <PatternLayout pattern="%d{yyyy-MM-dd'T'HH:mm:ss.SSSSSSSSSZ} [%t] %-5level %logger{0} - %msg%n" charset="UTF-8"/>
      <Policies>
        <!--<CronTriggeringPolicy schedule="0/5 * * * * ?"/>-->
        <!--<TimeBasedTriggeringPolicy interval="1"/>-->
       <SizeBasedTriggeringPolicy size="10000 MB"/>
      </Policies>
      <DefaultRolloverStrategy max="10"/>
    </RollingRandomAccessFile>
  </Appenders>
  <Loggers>
<!--
    <AsyncLogger name="*"  level="trace" includeLocation="true">
      <AppenderRef ref="File"/>
    </AsyncLogger>
-->
    <Root level="trace" includeLocation="false">
      <AppenderRef ref="File"/>
    </Root>
  </Loggers>
</Configuration>

```

The throughput performance results below:
Loggers all asynchronous,Throughput per thread in messages/second.

```
1 thread
274746 operations/second

2 threads 
116848 operations/second
113408 operations/second

4 threads
57831 operations/second
57201 operations/second
55981 operations/second
56746 operations/second
```

On windows11 (64bit) with OpenJDK11
CPU: AMD Ryzen 5 5600G@3.9GHz 6core,12thread.
MEM: 3600 32GB(16X2 )
SSD: WD SN570  512GB

Why is there such a big difference in my test results? Is it my configuration error that caused it?

https://logging.apache.org/log4j/2.x/manual/async.html

**Comment by tcmot** [comment]
@ppkarwasz 

Hi,Do you have any suggestions for me?

Thanks.

**Comment by ppkarwasz** [comment]
@tcmot,

You'll need to ask @remkop, he's the expert. Generally those figures seem a little bit low for me. On my 5 years old laptop I obtain around 700k ops/second with sync logging and a `File` appender.

All the figures on our website were obtained on UNIXes, which have a different file access semantics. Try using other appenders.

**Reply by remkop** [reply]
Yes seems low. 
These are the kind of numbers I expect to see with `includeLocation="true"`...

Can you double-check or run again after checking the logging configuration for includeLocation?

**Comment by tcmot** [comment]
@remkop  

I use includeLocation="false", I just want to test the RollingRandomAccess File Appender and Loggers all asynchronous.

Can I take a look at your log4j2.xml configuration file?
